### PR TITLE
Rewrite rendering flow to use glsl

### PR DESCRIPTION
**Pull request by sigvef**
Some nodes were removed completely in the process, either because of
large dependencies or just because they turned out to not be useful.
Specifically, SobelNode was removed only because of the external
dependencies, so it should be reimplemented in glsl by someone.

Rendering is now deferred to a separate render loop to improve
performance and UI snappiness.

### CODE REVIEW

**Comment by stianjensen** [comment]
Requests to the deleted libs should be removed from index.html

(opencv, perlin, texturegen-core)

**Comment by stianjensen** [comment]
Polar hasn't been removed from the list of nodes in the UI

**Comment by stianjensen** [comment]
It seems this has stopped disconnecting inputs from rerendering again. (May be fixed by setting the node to dirty in the disconnect method?)

**Comment by stianjensen** [comment]
I'm getting an error in the console when moving nodes:

```
interact.js:184 Uncaught TypeError: Cannot read property 'options' of null
```

File: src/CircularSineNode.js [hunk 1]
```diff
@@ -25,24 +42,17 @@
           step: 1,
           default: 5
         })
-      ]);
+      ], shader);
     }
 
     render() {
```
Comment (stianjensen) [inline src/CircularSineNode.js]: Why does this node still have a render method?

File: src/THREEPreviewNode.js [hunk 2]
```diff
@@ -1,6 +1,17 @@
 (function(TextureGen, THREE) {
   'use strict';
 
+var shader = `
```
Comment (stianjensen) [inline src/THREEPreviewNode.js]: What's this shader used for?

Since THREEPreviewNode extends BaseNode, it doesn't seem like it's used for anything.

**Comment by sigvef** [comment]
ping

File: src/GradientValue.js [hunk 3]
```diff
@@ -20,6 +24,17 @@
...
 
+    getOutput() {
+      this.x0 = 0;
+      this.y0 = 0;
+      this.x1 = 256;
+      this.y1 = 0;
+      var gradient = this.toCanvasStyle();
+      this.ctx.fillStyle = gradient;
+      this.ctx.fillRect(0, 0, 256, 1);
+      return this.canvas;
```
Comment (stianjensen) [inline src/GradientValue.js:35]: I guess this wrecks export all over again, but I'm not sure how we want to combat that.

**Comment by stianjensen** [comment]
```
[.CommandBufferContext]GL ERROR :GL_INVALID_OPERATION : glUniform1fv: wrong uniform function for type
```

Errors in glsl are hard to track down, but I'm getting this message in my console, but it seems to be related to the pattern node.
I also see that the pattern node can't generate brick pattern for me, so it might be related to that.

File: src/GradientValue.js (hunk 3 above)
Comment (sigvef) [inline src/GradientValue.js:35]: image

I think exporting works fine, afaict.

**Comment by sigvef** [comment]
ping

**Comment by sigvef** [comment]
Outdated now.

**Pull request closed without merging**